src/main/kotlin/com/hypto/iam/server/authProviders/WorkOSAuthProvider.kt

```diff
@@ -0,0 +1,31 @@
+package com.hypto.iam.server.authProviders
+
+import com.hypto.iam.server.ROOT_ORG
+import com.hypto.iam.server.configs.AppConfig
+import com.hypto.iam.server.security.AuthMetadata
+import com.hypto.iam.server.security.OAuthUserPrincipal
+import com.hypto.iam.server.security.TokenCredential
+import com.workos.WorkOS
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+object WorkOSAuthProvider : BaseAuthProvider("workos"), KoinComponent {
+    private val appConfig: AppConfig by inject()
+
+    override fun getProfileDetails(tokenCredential: TokenCredential): OAuthUserPrincipal {
+        val workos = WorkOS(appConfig.workOS.secretKey)
+        require(!tokenCredential.value.isNullOrEmpty()) {
+            "Token credential value cannot be null or empty"
+        }
+        val profileAndToken = workos.sso.getProfileAndToken(tokenCredential.value, appConfig.workOS.clientId)
+        return OAuthUserPrincipal(
+            tokenCredential,
+            ROOT_ORG,
```

Comment:
Why does userPrincipal have ROOT_ORG as org_id? Can we explore ways to store org_id in workOS app level metadata at the time of connection?

Reply:
As discussed, there is no way to store org_id in workOS app level metadata at the time of connection. Hence keeping org_id as ROOT_ORG